docs/docs/getting_started/usage-and-installation.mdx

```diff
@@ -0,0 +1,734 @@
+---
+title: Installation and Usage
+---
+
+## Installation
+
+### Dependencies
+
+1. You will need [z3](https://github.com/Z3Prover/z3) installed to use Warp.
+
+- Install command on macOS:
+
+```
+brew install z3
+```
+
+- Install command on Ubuntu:
+
+```
+sudo apt install libz3-dev
+```
+
+2. Have Python 3.7 installed with the virtualenv ([`venv`](https://docs.python.org/3/library/venv.html)) module in your base env.
+
+### Warp Installation Method 1
+
+Without any virtual environment activated perform the following in order:
+
+1. Add the warp package from npm.
+
+```
+yarn global add @nethermindeth/warp
+```
+
+2. Ensure the package was added by checking the version number:
+
+```
+warp version
+```
+
+3. Install the dependencies:
+
+```
+warp install
+```
+
+4. Test the installation worked by transpiling an example ERC20 contract:
+
+```
+warp transpile example_contracts/ERC20.sol
+```
+
+### Warp Installation Method 2 (from source/for devs)
+
+Make sure you have the [dependencies](#dependencies) installed first.
+
+With a virtual environment (recommended Python3.7) activated:
+
+1. Clone this repo and change directory into the `warp` folder.
+
+2. Install the JavaScript dependencies:
+
+```
+yarn
+```
+
+3. Install the Python dependencies:
+
+```
+pip install -r requirements.txt
+```
+
+If you are using a M1 chipped Mac and getting a `'gmp.h' file not found` error when installing Cairo run the following:
+
+```
+CFLAGS=-Ibrew --prefix gmp/include LDFLAGS=-Lbrew --prefix gmp/lib pip install ecdsa fastecdsa sympy
+```
+
+Then run the pip command above again.
+
+4. Compile the project:
+
+```
+yarn tsc
+yarn warplib
+```
+
+5. Test the installation worked by transpiling an example ERC20 contract:
+
+```
+bin/warp transpile example_contracts/ERC20.sol
+```
+
+## Usage
+
+Warp 2 allows transpiling directly from Solidity to Cairo and transpiled .cairo file can be deployed to starknet.
+
+### Transpiling
+
+To transpile a solidity file, run:
+
+```
+warp transpile [PATH_TO_SOLIDITY_FILE]
+```
+
+For example:
+
+```
+warp transpile contracts/token/ERC20/ERC20.sol
+```
+
+The output of the transpilation could be found in warp_output folder.
+
+Now,the file can be compiled and deployed. Run the following to compile the Cairo contract:
+
+```
+warp compile warp_output/contracts/token/ERC20/ERC20__WARP_CONTRACT__ERC20.cairo
+```
+
+### Deploying
+
+```
+warp deploy [PATH_TO_TRANSPILED_CAIRO_CONTRACT] --use_cairo_abi --inputs  [INPUT_ARGUMENTS]
+```
+
+For example:
+
+```
+warp deploy warp_output/contracts/token/ERC20/ERC20__WARP_CONTRACT__ERC20.cairo --use_cairo_abi --inputs  10,0x4e,0x45,0x54,0x48,0x45,0x52,0x43,0x4f,0x49,0x4e,4,0x4e,0x45,0x54,0x48
```

Comment:
Can you use examples without the --use_cairo_abi, it would be easier for the user to understand if it uses the Solidity ABI